article/app/views/fragments/recipeArticleBody.scala.html

```diff
@@ -0,0 +1,174 @@
+@recipePage(recipe: model.content.RecipeAtom, i: Int) = {
+    <div data-recipe-index="recipe__@i" id="@recipe.atom.id" class="js-recipe__article--structured recipe__article--structured" itemtype="http://schema.org/Recipe">
+        <div class="js-recipe__article--structured-headline recipe__article--structured-headline">
+            <h1 itemprop="name">@recipe.data.title</h1>
+        </div>
+
+        <div class="recipe__article--structured-meta">
+            <div class="structured-meta">
+                @defining(recipe.data.serves){ maybeServes =>
+                    @for(serves <- maybeServes) {
+                        <span class="rounded-icon structured-meta__circle">@serves.from</span>
+                        @if(serves.from != serves.to) {to<span class="rounded-icon structured-meta__circle">@serves.to</span> }
+                        @yieldServingType(serves)
+                    }
+                }
+                @if(totalTime(recipe)) {<span class="rounded-icon structured-meta__circle">@totalTime(recipe)</span>hours }
+            </div>
+        </div>
+
+        <div class="recipe__article--structured-ingredients">
+        @for(list <- recipe.data.ingredientsLists) {
+            <ul class="ingredients-list">
+                <li class="ingredients-title">@if(list.title) {@list.title} else {Ingredients}</li>
+                @for(ingredient <- list.ingredients) {
+                    <li itemprop="recipeIngredient">@ingredientValue(ingredient)</li>
+                }
+            </ul>
+        }
+        </div>
+
+        <div class="recipe__article--structured-method">
+            <ul class="method-list">
+                @recipe.data.steps.zipWithIndex.map{case(step, i) => <li><span class="number">@{i + 1}</span>@step</li> }
+            </ul>
+        </div>
+    </div>
+}
+
+@totalTime(recipe: model.content.RecipeAtom) = @{
+    (recipe.data.time.preparation ++ recipe.data.time.cooking).map(_.toInt).reduceOption(_ + _)
+}
+
+
+@yieldServingType(serves: com.gu.contentatom.thrift.atom.recipe.Serves) = @{
+    serves.`type` match {
+        case "serves" => "servings"
+        case "makes" => s"${serves.unit}"
+        case "quantity" => "portions"
+    }
+}
+
+@ingredientValue(ingredient: com.gu.contentatom.thrift.atom.recipe.Ingredient) = @{
+    val q = ingredient.quantity
+    .map(formatQuantity)
+    .orElse(ingredient.quantityRange.map(range => s"${formatQuantity(range.from)}-${formatQuantity(range.to)}" ))
+    .getOrElse("")
+    s"""${q} ${formatUnit(ingredient.unit.getOrElse(""))} ${ingredient.item}"""
+}
+
+@formatUnit(unit: String) = @{
+    unit match {
+    case "dsp" => "dessert spoon"
+    case "tsp" => "teaspoon"
+    case "tbsp" => "tablespoon"
+    case _ => unit
+    }
+}
+
+@formatQuantity(q: Double) = @{
+    q match {
+    case qty if qty == qty.toInt => qty.toInt.toString
+    case 0.75 => "¾"
+    case 0.5 => "½"
+    case 0.25 => "¼"
+    case 0.125 => "⅛"
+    case _ => q.toString
+    }
+}
```

Comment:
We already have very similar formatting functions in https://github.com/guardian/frontend/blob/master/common/app/views/fragments/atoms/structureddata/recipe.scala.html#L76-L94

What about extracting all the formatting logic in their own `RecipeFormatter` or `Recipe` object, so they can be reused everywhere. That would help staying consistent across the different pages.

Also I think Twirl template code blocks should only be returning UI elements, the same way React components do 😬